scenes/cslib_array.txt

```diff
@@ -247,6 +247,180 @@
 *return
 
 
+*comment TO_STRING
+*comment ------------------------------
+*comment Returns all elements of an array joined as a string.
+*comment An optional string may be passed with which
+*comment to separate the elements.
+*comment ------------------------------
+*comment    params:
+*comment        p_array_ref(string): the name of the array to convert to a string
+*comment        p_separator(string)?: the string to insert between each array element
```

Comment:
does the question mark means optional? if so, please specify

Reply:
Yeah, I was trying to think a non-verbose way of indicating that. I think I've used '?' in another spot or two. I can add the words optional to the description as well.

Reply:
Maybe it would be good to settle on some general conventions and include an example in the readme explaining how to read the docs.

Reply:
I think when we review the contribution guide, we could cover that. Probably be worth having a style guide as well.